===== Patch instructions generated by SAVER =====
[+] { Insert tmp := try:cache_ent->next at 63 (line 201, column 7) and replace the original expression at 68 (line 216, column 4) under true }
===== Actual patch in diff format =====
@@ -200,6 +200,7 @@ try (struct search_ctx *ctx)
       break;
   if (cache_ent)
     {
+      tmp = cache_ent->value;
       ctx->is_cache = 1;
       if (iterate_device (cache_ent->value, ctx))
 	{
@@ -213,7 +214,7 @@ try (struct search_ctx *ctx)
 	  grub_free (cache_ent->key);
 	  grub_free (cache_ent->value);
 	  grub_free (cache_ent);
-	  *prev = cache_ent->next;
+	  *prev = tmp;
 	}
     }
 
